src/clr-angular/forms/common/layout.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directive, ElementRef, Input, Renderer2 } from '@angular/core';
+import { Layouts, LayoutService } from './providers/layout.service';
+
+@Directive({
+  selector: '[clrLayout]',
+})
+export class ClrLayout {
+  @Input('clrLayout') layout: Layouts;
+
+  constructor(private el: ElementRef, private renderer: Renderer2, private layoutService: LayoutService) {}
+
+  ngAfterContentInit() {
+    this.renderer.addClass(this.el.nativeElement, 'clr-form-' + this.layoutService.layout);
+  }
+
+  ngOnInit() {
+    if (this.layout) {
+      this.layoutService.layout = this.layout;
```

Comment:
Should we be able to change the layout from horizontal to vertical or vice-versa using the input? Here we are initializing the layout once.

Reply:
This does seem like something you want to do once, and because changing the layout is a jarring experience for the user. Eudes and I talked about it briefly, and I believe we agreed on that point but happy to debate it further.

I'll add some check to ensure the input value is valid.